### PR TITLE
factory: populate /etc/locale.conf with systemd build-time setting

### PR DESCRIPTION
**Pull request by lucab**
This adds `/etc/locale.conf` to the set of configuration files
populated by tmpfiles.d factory `/etc` handling.
In particular, the build-time locale configuration in systemd is
now wired to a /usr factory file, and installed to the system.
On boot, if other locale customization tools did not write
`/etc/locale.conf` on the system, the factory default file gets
copied to `/etc` by systemd-tmpfiles.
This is done in order to avoid skews between different system
components when no locale settings are configured. At that point,
systemd can safely act as the fallback owner of `/etc/locale.conf`.

Closes: https://github.com/systemd/systemd/issues/21723

### CODE REVIEW

**Comment by keszybz** [comment]
https://sourceware.org/bugzilla/show_bug.cgi?id=17318 added C.UTF-8 to glibc, which is nice because we can make use of C.UTF-8 without any qualms.

Looks good to me.

File: factory/templates/locale.conf.in [hunk 1]
```diff
@@ -0,0 +1 @@
+LANG="{{ SYSTEMD_DEFAULT_LOCALE }}"
```
Comment (bluca) [inline factory/templates/locale.conf.in:3]: Do we need a license header? Other config files we ship have them - also maybe a comment ` # File generated by ...`? Mine generated from Debian has a similar comment

Comment (lucab) [reply factory/templates/locale.conf.in:3]: I was unsure whether other parsers support comments here. I checked Fedora and the [systemd docs](https://www.freedesktop.org/software/systemd/man/locale.conf.html#Example) and didn't see any comment (on a Debian system I don't have such file, what generated it?). `systemd-firsboot` as well does not seem to inject any comment.

If we feel it is safe to have comments in there, I can add the generator/owner. A license is not needed IMHO, there is no alternative arrangement/approach for these few bytes really.

Comment (keszybz) [reply factory/templates/locale.conf.in:3]: I think it's fine not to add the header here. This file is likely to be modified by adding more lines, by systemd-localed or even by some image modification scripts, and then the comment that this file was "created by systemd" will be silly.

Comment (lucab) [reply factory/templates/locale.conf.in:3]: Oh woops, conflicting push from my side then. On the second commit, should I keep the docs changes or throw the whole commit out?

Comment (keszybz) [reply factory/templates/locale.conf.in:3]: Hmm, maybe say something like "This is a fallback locale.conf file provided by systemd." ?

Comment (lucab) [reply factory/templates/locale.conf.in:3]: Ack, reworded in a similar way.

**Review by bluca — APPROVED** [review]

**Pull request merged**